ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Abstract.hs

```diff
@@ -185,7 +193,9 @@ class ( Show (ChainState    p)
                 -> m (Maybe (IsLeader p))
 
   -- | Apply a block
-  applyChainState :: SupportedBlock p b
+  --
+  -- TODO this will only be used with headers
```

Comment:
Let's create a ticket to rename this.

Reply:
Let's also create a ticket to introduce the `BlockHeader` type family and use it throughout the consensus code base.